### DISCUSSION TITLE
Commercial Version of Amazing Oqtane Studio on Oqtane Market Place

### DISCUSSION THREAD

**Discussion by vnetonline** [oqtane/oqtane.framework#4378] [Q&A]
@sbwalker as you are aware I have created the [Amazing Oqtane Studio (AOS)](https://amazingoqtanestudio.com) SaaS Solution. 

I would like to make the AOS Pro Developer and AOS Expert Developer to be purchased from Oqtane Marketplace on a subscription.

I would like to use the webhook to then create the user account with appropriate roles so that the AOS Developer can gain access to [AOS Dashboard](https://dashboard.amazingoqtanestudio.com)

I see that when you create a product on Oqtane Marketplace there is an option to select `No Download`.

Can you please explain how I can offer this commercial product on the Oqtane Marketplace with no download and give the purchaser access to AOS SaaS Solution?

I also need the ability to disable their account if they fail to pay the subscription or cancel their subscription. I rather not roll my own payment gateway.

any guidance you are able to provide will be greatly appreciated :)

**Comment by sbwalker** [comment]
@vnetonline this repo is intended for items related to the open source framework. Marketplace items should be submitted here: https://github.com/oqtane/Oqtane.LicensedModule

The Marketplace was designed for downloadable products in an App Store delivery model - not SaaS products (I believe I already explained this to you in an email you sent related to the same topic). 

The No Download option is intended for products where there is nothing to download however I would need to review the logic to determine the expected behavior. 

Since the Marketplace was never designed for your scenario I expect it is missing some features. These features could be added in the future however there are other priorities at the moment which are higher.

**Reply by vnetonline** [reply]
@sbwalker apologies but I never got an email from you in regards to this topic

**Reply by sbwalker** [reply]
Perhaps it went to your Junk folder...

Sent: Tuesday, May 28, 2024 9:12 AM
To: Vineet Belani
Subject: RE: Amazing Oqtane Studion - Marketing place Commercial Product on Subscription 

Hi Vineet,

The Oqtane Marketplace supports subscriptions for downloadable/licensed products (ie. it is an App Store). It does not support subscriptions for remotely hosted SaaS products.

Thank you!

**Reply by vnetonline** [reply]
Okay never got that email ... thanks for posting it here
